Scripts/Commands: Fix guid usage and enhance commands

Rochet2 opened the pull request:
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

This is a cherry pick of https://github.com/TrinityCore/TrinityCore/pull/18278 and is needed to finish up https://github.com/TrinityCore/TrinityCore/issues/18863

**Changes proposed:**

- Fix targeting gameobjects and creatures for .dist and .gps. The old code used DB guid to search from core generated guid lists.
- Fix some error messages telling no player found even if creatures and gameobjects and players were searched
- Change atoi to atoul when receiving lowguids as string or other uint32 values
- Use ObjectGuid::LowType instead of uint32 for lowguid type
- Allow .gob info to take in gameobject guid link as well as entry link in addition to gameobject entry and update the documentation of the command
- Change .gob delete to post the spawnid (dbguid) instead of the ingame generated guid (which is basically never seen or used by the player in any other commands that I see)
- Allow spawntimesecs to be negative when spawning a gameobject
- Disable searching by ingame spawn id in commands. (it makes little sense for a user to provide a number which is then attempted to be used as DB guid and non DB guid when both could have a match)
- Update .gob turn documentation to tell about possibility to provide orientation in command
- Correct the types in .gob add command result trinity_string
- Correct the creature search for .npc del <lowguid>
- Change .event <id> to .event info <id>
- Add .go offset command
- Remove the .go command for empty string (no subcommand given) because it just used .go xyz
- Extend .gob info to show size, faction, flags and model dimensions
- Remove some unnecessary casts
- Document .wp show better (first and last were not documented)
- Insert spawnid to .wp show info error message
- Fix spawntime printing for .gobject target
- Fix guid targetting for .npc set movetype <guid> <type>
- Fix query by spawnid for .wp show info
- Fix deleting of existing waypoints when twice doing the command .wp show on
- Fix deleting of existing waypoints when doing the command .wp show off
- Change wpguid column in DB and in core to uint32, which is what the spawnid type is in core and db.

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)
Closes https://github.com/TrinityCore/TrinityCore/issues/18241
Closes https://github.com/TrinityCore/TrinityCore/issues/16733

**Tests performed:** (Does it build, tested in-game, etc.)

- [x] Does it build?

**Known issues and TODO list:** (add/remove lines as needed)

Rochet2 committed:
Scripts/Commands: Fix guid usage and enhance commands

- Fix targeting gameobjects and creatures for .dist and .gps. The old code used DB guid to search from core generated guid lists.
- Fix some error messages saying 'no player found' even if creatures and gameobjects and players were searched
- Change atoi to atoul when receiving lowguids as string or other uint32 values
- Use ObjectGuid::LowType instead of uint32 for lowguid type
- Allow .gob info to take in gameobject guid link as well as entry link in addition to gameobject entry and update the documentation of the command
- Change .gob delete to post the spawnid (dbguid) instead of the ingame generated guid (which is basically never seen or used by the player in any other commands that I see)
- Allow spawntimesecs to be negative when spawning a gameobject
- Disable searching by ingame spawn id in commands. (it makes little sense for a user to provide a number which is then attempted to be used as DB guid and non DB guid when both could have a match)
- Update .gob turn documentation to tell about possibility to provide orientation in command
- Correct the types in .gob add command result trinity_string
- Correct the creature search for .npc del
- Change .event to .event info
- Add .go offset command
- Remove the .go command for empty string (no subcommand given) because it just used .go xyz
- Extend .gob info to show size, faction, flags and model dimensions
- Remove some unnecessary casts
- Document .wp show better (first and last were not documented)
- Insert spawnid to .wp show info error message
- Fix spawntime printing for .gobject target
- Fix guid targetting for .npc set movetype
- Fix query by spawnid for .wp show info
- Fix deleting of existing waypoints when twice doing the command .wp show on
- Fix deleting of existing waypoints when doing the command .wp show off
- Change wpguid column in DB and in core to uint32, which is what the spawnid type is in core and db.

 Conflicts:
	sql/base/auth_database.sql
	src/server/game/Accounts/RBAC.h
	src/server/game/Chat/Chat.cpp
	src/server/scripts/Commands/cs_go.cpp
	src/server/scripts/Commands/cs_gobject.cpp
	src/server/scripts/Commands/cs_misc.cpp
	src/server/scripts/Commands/cs_npc.cpp
	src/server/scripts/Commands/cs_wp.cpp

Commit:
Update RBAC.h

Sync PR branch file with TC master

Commit:
Merge branch 'master' into fixcommands2

Rochet2 commented on src/server/game/Accounts/RBAC.h:
```diff
@@ -755,7 +755,8 @@ enum RBACPermissions
     RBAC_PERM_COMMAND_RELOAD_SCENE_TEMPLATE                  = 850,
-    RBAC_PERM_COMMAND_GO_OFFSET                              = 851,
+    RBAC_PERM_COMMAND_RELOAD_AREATRIGGER_TEMPLATE            = 851,
+    RBAC_PERM_COMMAND_GO_OFFSET                              = 852,
```
hmm, need to fix the SQLs for this too before merge

Rochet2 committed:
Fix merge

Commit:
Rename xxxx_xx_xx_xx_auth.sql to 2017_99_99_00_auth.sql

Commit:
Rename xxxx_xx_xx_x0_world.sql to 2017_99_99_18978_world.sql

Rochet2 commented:
Note me if I need to do something about this before merging.

Merging this will allow us to fix the master/3.3.5 rbac permission desync (before it gets any worse) and continue to work on fixing commands for both versions.

Aokromes commented:
there is no desync anymore.

Rochet2 commented:
I see, it was fixed by ariel- on https://github.com/TrinityCore/TrinityCore/commit/9e65fead38c486ec8a971a09a95ffae912943a83

Shauren commented on src/server/scripts/Commands/cs_gobject.cpp:
```diff
@@ -373,7 +363,7 @@ class gobject_commandscript : public CommandScript
             if (!owner || !ownerGuid.IsPlayer())
             {
-                handler->PSendSysMessage(LANG_COMMAND_DELOBJREFERCREATURE, ownerGuid.ToString().c_str(), object->GetGUID().ToString().c_str());
+                handler->PSendSysMessage(LANG_COMMAND_DELOBJREFERCREATURE, ownerGuid.GetCounter(), guidLow);
```
bug here (both in existing code and proposed)
db entry 274
>Game Object (GUID: %u) is referenced by nonexistent creature %u in GO list, can't be deleted.

counter/lowguid is uint64, %u is not sufficient
always prefer printing guid using ToString(), not its low part

Rochet2 replied:
This is fixed by the other PR that tackles the types in general and fixes most if not all such differences in commands. https://github.com/TrinityCore/TrinityCore/pull/18863

However it uses tostring on the lowguid instead of the full guid. I guess in this case the correct way is as you say to use the tostring because its not a gameobject link.

But in general most of the code for commands gives out the lowguids instead of full guids since the commands use lowguid to refer to objects.
Hmm.

Rochet2 committed:
Remove edits to LANG_COMMAND_DELOBJREFERCREATURE

Rochet2 committed:
Merge branch 'master' into fixcommands2

Commit:
Travis temp fix local to PR

- add  2017_99_99_00_auth.sql to auth.updates in base/auth_database.sql
- also add changes in rbac_permissions + rbac_linked_permissions

tkrokli commented:
Any other changes needed in this PR before merging?

Rochet2 committed:
Remove wpguid column change - its already bigint unsigned

Rochet2 commented:
Couldnt find other things to do myself atm.
